src/Language/PureScript/ModuleDependencies.hs

```diff
@@ -71,8 +71,10 @@ usedModules ams d =
     [mn]
   forDecls (FixityDeclaration fd)
     | Just mn <- extractQualFixity fd, mn `notElem` ams = [mn]
-  forDecls (TypeInstanceDeclaration _ _ (Qualified (Just mn) _) _ _)
-    | mn `notElem` ams = [mn]
+  forDecls (TypeClassDeclaration _ _ cs _)
+    = filter (`notElem` ams) (constraintModules cs)
+  forDecls (TypeInstanceDeclaration _ cs (Qualified (Just mn) _) _ _)
+    = filter (`notElem` ams) (mn : constraintModules cs)
```

Comment:
You mean extraneous imports are a warning? I don't think that's a problem - I was talking about the fact that names are unresolvable when they're imported now, so we might have some extra dependencies in the graph if we only consider imports, but the advantage would be we can delete almost all of this code here if we do just rely on imports instead.